tests/v2/test_wakunode.nim

```diff
@@ -636,6 +687,108 @@ procSuite "WakuNode":
+                                  memIndexOpt= memIndexOpt2, 
+                                  onchainMode = false, 
+                                  pubsubTopic = rlnRelayPubSubTopic)
+      await node2.start()
+
+      # node 3
+      node3.mountRelay(@[rlnRelayPubSubTopic])
+      let (groupOpt3, memKeyPairOpt3, memIndexOpt3) = rlnRelaySetUp(3) # set up rln relay inputs
+      # mount rlnrelay in off-chain mode
+      waitFor node3.mountRlnRelay(groupOpt = groupOpt3, 
+                                  memKeyPairOpt = memKeyPairOpt3, 
+                                  memIndexOpt= memIndexOpt3, 
+                                  onchainMode = false, 
+                                  pubsubTopic = rlnRelayPubSubTopic)
+      await node3.start()
+
+      # connect them together
+      await node1.connectToNodes(@[node2.peerInfo])
+      await node3.connectToNodes(@[node2.peerInfo])
+      
+      # define a custom relay handler
+      var completionFut = newFuture[bool]()
+      proc relayHandler(topic: string, data: seq[byte]) {.async, gcsafe.} =
+        let msg = WakuMessage.init(data)
+        if msg.isOk():
+          let val = msg.value()
+          debug "The received topic:", topic
+          if topic == rlnRelayPubSubTopic:
+            completionFut.complete(true)
+
+      # mount the relay handler
+      node3.subscribe(rlnRelayPubSubTopic, relayHandler)
+      await sleepAsync(2000.millis)
+
+      # prepare the message payload
+      var payload {.noinit.}: array[32, byte]
+      for x in payload.mitems: x = 1
+
+      # prepare the epoch
+      var epoch {.noinit.}: Epoch
+      for x in epoch.mitems: x = 2
+
+      # prepare the proof
+      let nonSpamProofRes = node1.wakuRlnRelay.rlnInstance.proofGen(data = payload, 
+                                                                memKeys = node1.wakuRlnRelay.membershipKeyPair, 
+                                                                memIndex = MembershipIndex(4), 
+                                                                epoch = epoch)
+      doAssert(nonSpamProofRes.isSome)
+      let nonSpamProof = nonSpamProofRes.value     
+
+      let message = WakuMessage(payload: @payload, 
+                                contentTopic: contentTopic,  
+                                proof: nonSpamProof)
+
+
+      ## node1 publishes a message with an invalid non-spam proof, the message is then relayed to node2 which in turn 
+      ## attempts to verify the non-spam proof and fails hence does not relay the message to node3, thus the relayHandler of node3
+      ## never gets called
+      ## verification at node2 occurs inside a topic validator which is installed as part of the waku-rln-relay mount proc
+      await node1.publish(rlnRelayPubSubTopic, message)
+      await sleepAsync(2000.millis)
+
+      check:
+        # the relayHandler of node3 never gets called
+        (await completionFut.withTimeout(10.seconds)) == false
```

Comment:
Is it necessary to wait a full `10` seconds? Just thinking that these long waits in the CI is starting to add up.

Reply:
> BTW, RLN-Relay tests are not yet effective in our CI checks :) they are conditioned to the RLN flag which is set to false by default

Can we instrument this somehow? Perhaps set RLN to true in test context or so?

Reply:
I am not sure how to selectively enable rln tests i.e., active on GH CIs but not for local builds. I am going to track it down in this issue https://github.com/status-im/nim-waku/issues/741